tests/functional/aws-node-sdk/test/object/deleteMpu.js

```diff
@@ -2,28 +2,48 @@ import assert from 'assert';
 
 import withV4 from '../support/withV4';
 import BucketUtility from '../../lib/utility/bucket-util';
+import config from '../../../../../lib/Config';
 
 const bucket = 'functestabortmultipart';
 const key = 'key';
 
 const itSkipIfAWS = process.env.AWS_ON_AIR ? it.skip : it;
 
+const westLocation = config.locationConstraints ? 'scality-us-west-1'
+: 'us-west-1';
+const eastLocation = config.locationConstraints ? 'aws-us-east-1' : 'us-east-1';
+const describeSkipIfOldConfig = config.locationConstraints ? describe :
+describe.skip;
+
+// Why are we skipping error 404 if old config?
+// AWS returns 404 - NoSuchUpload in us-east-1. This behavior
+// can be toggled to be compatible with AWS by enabling
+// usEastBehavior in the config.
+const confLocations = [
+  { name: 'us-west-1', statusCode: 204, location: westLocation, describe },
+  { name: 'us-east-1', statusCode: 404, location: eastLocation,
+    describe: describeSkipIfOldConfig },
+];
+
 describe('DELETE multipart', () => {
     withV4(sigCfg => {
         const bucketUtil = new BucketUtility('default', sigCfg);
         const s3 = bucketUtil.s3;
 
-        function _assert204StatusCode(uploadId, callback) {
+        function _assertStatusCode(uploadId, statusCode, callback) {
             const request =
             s3.abortMultipartUpload({ Bucket: bucket, Key: key,
                 UploadId: uploadId }, err => {
                 const statusCode =
```

Comment:
here I think you are overwriting the `statusCode` from the parameters, you may want to name them differently